File: eTouristAgencyAPI/eTouristAgencyAPI.Services/Database/Models/Hotel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace eTouristAgencyAPI.Services.Database.Models;

[Table("Hotel")]
public partial class Hotel
{
    [Key]
    public Guid Id { get; set; }

    [StringLength(255)]
    public string Name { get; set; } = null!;

    public Guid CityId { get; set; }

    public DateTime CreatedOn { get; set; }

    public DateTime ModifiedOn { get; set; }

    public int StarRating { get; set; }

    public Guid CreatedBy { get; set; }

    public Guid ModifiedBy { get; set; }

    [ForeignKey("CityId")]
    [InverseProperty("Hotels")]
    public virtual City City { get; set; } = null!;

    [ForeignKey("CreatedBy")]
    [InverseProperty("HotelCreatedByNavigations")]
    public virtual User CreatedByNavigation { get; set; } = null!;

    [InverseProperty("Hotel")]
    public virtual ICollection<HotelImage> HotelImages { get; set; } = new List<HotelImage>();

    [ForeignKey("ModifiedBy")]
    [InverseProperty("HotelModifiedByNavigations")]
    public virtual User ModifiedByNavigation { get; set; } = null!;

    [InverseProperty("Hotel")]
    public virtual ICollection<Offer> Offers { get; set; } = new List<Offer>();
}

File: Frontend/etouristagency_mobile/lib/models/hotel/hotel.dart
import 'package:etouristagency_mobile/models/city/city.dart';
import 'package:etouristagency_mobile/models/hotel/hotel_image.dart';

class Hotel {
  String? id;
  String? name;
  int? starRating;
  String? cityId;
  City? city;
  List<HotelImage>? hotelImages;

  Hotel(this.id, this.name, this.starRating, this.cityId, this.city, this.hotelImages);

  factory Hotel.fromJson(Map<String, dynamic> json) {
    return Hotel(
      json["id"],
      json["name"],
      json["starRating"],
      json["cityId"],
      json["city"] != null ? City.fromJson(json["city"]) : null,
      json["hotelImages"] != null ? (json["hotelImages"] as List).map((x)=> HotelImage.fromJson(x)).toList() : null
    );
  }
}